framework/helpers/BaseStringHelper.php

```diff
@@ -338,4 +338,35 @@ public static function base64UrlDecode($input)
     {
         return base64_decode(strtr($input, '-_', '+/'));
     }
+
+    /**
+     * Masks a token to make it uncompressible.
```

Comment:
Applies random mask to a token and prepends mask used to the result making the string always unique.

Used to mitigate BREACH attack.

@see https://blog.ivanristic.com/2013/08/defending-against-the-breach-attack.html